src/coreclr/jit/compiler.h

```diff
@@ -8692,11 +8692,28 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #elif defined(TARGET_ARM64)
         return FP_REGSIZE_BYTES;
 #else
-        assert(!"maxSIMDStructBytes() unimplemented on target arch");
+        assert(!"getMaxVectorByteLength() unimplemented on target arch");
         unreached();
 #endif
     }
 
+    //------------------------------------------------------------------------
+    // getPreferredVectorByteLength: Gets the preferred length, in bytes, to use for vectorization
+    //
+    uint32_t getPreferredVectorByteLength() const
+    {
+#if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_XARCH)
+        uint32_t preferredVectorByteLength = opts.preferredVectorByteLength;
+
+        if (preferredVectorByteLength != 0)
```

Comment:
should it also block 384 and > 512? (esp if you plan on leaving this flag in long term)

Reply:
This is a "maximum allowed" and so if a user says `DOTNET_PreferredVectorBitWidth=2048` we'll just pick the largest supported size less than or equal to that. Same with `384`.

This will end up working well with Arm64 SVE as well, where it allows multiples of 128 up to 2048 to be provided by hardware.